src/api/auth.js

```diff
@@ -174,6 +177,13 @@ class AuthInterface {
   * ])
   */
   genShareMDataUri(permissions) {
+    if (!permissions) {
```

Comment:
I think this validation is redundant with the one done within `validateShareMDataPermissions`, so we should be able to remove it form here. Also, the validation underneath is probably also better to have it within the `validateShareMDataPermissions`.